error: invalid instruction, mffprwz ...

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

Please:

* Read the documentation to test with the latest developer build.
* Check if other person has already created the same issue to avoid duplicates. You can comment on it if there already is an issue.
* Try to be as detailed as possible in your report.
* Report only one problem per created issue.


This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => master, 834c99255320fb565259d3e9177edcb590d4a6be
- Operating System / Platform => Ubuntu 18.04, Power 9
- Compiler => clang version 10.0.0 (git@github.com:llvm/llvm-project.git 2f51a43f8c2b117e69a45eae6a7c849e035a793b)

##### Detailed description
```
In file included from .../opencv/modules/core/src/arithm.cpp:50:
In file included from .../opencv/modules/core/src/precomp.hpp:49:
In file included from .../opencv/modules/core/include/opencv2/core/utility.hpp:56:
In file included from .../opencv/modules/core/include/opencv2/core.hpp:57:
In file included from .../opencv/modules/core/include/opencv2/core/matx.hpp:54:
In file included from .../opencv/modules/core/include/opencv2/core/saturate.hpp:49:         
.../opencv/modules/core/include/opencv2/core/fast_math.hpp:200:5: error: invalid instruction
    CV_INLINE_ROUND_DBL(value);                      
    ^                                                
.../opencv/modules/core/include/opencv2/core/fast_math.hpp:105:40: note: expanded from macro 'CV_INLINE_ROUND_DBL'
        __asm__( "fctiw %[temp],%[in]\n\tmffprwz %[out],%[temp]\n\t" : [out] "=r" (out), [temp] "=d" (temp) : [in] "d" ((double)(value)) : ); \
                                       ^
<inline asm>:2:2: note: instantiated into assembly here
        mffprwz 4,0                                                                                                                             
        ^~~~~~~
```
After investigating the most recent [Power ISA](https://openpowerfoundation.org/?resource_lib=power-isa-version-3-0) (March 29, 2017. Page 131 of the pdf or 113 of the document) `mffprwz` is a mnemonic for `mfvsrwz`. Specifically: 
```
Extended Mnemonics        Equivalent To
mffprwz      RA,FRS       mfvsrwz      RA,FRS
mfvrwz       RA,VRS       mfvsrwz      RA,VRS+32
```

After investigating clang, it _does not_ implement `mffprwz`, instead only `mfvsrwz`. I will be passing this along to the llvm project as well but we can fix this here as well, probably faster. I've prepared a patch to follow as well.

##### Steps to reproduce
 1. Checkout opencv.
 1. Make build directory.
 1. `cmake <path to source> -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_COMPILER=clang`
 1. `make -j###`
 1. Receive above error message.

--- a/modules/core/include/opencv2/core/fast_math.hpp
+++ b/modules/core/include/opencv2/core/fast_math.hpp
@@ -102,7 +102,7 @@
     #define CV_INLINE_ROUND_DBL(value) \
         int out; \
         double temp; \
-        __asm__( "fctiw %[temp],%[in]\n\tmffprwz %[out],%[temp]\n\t" : [out] "=r" (out), [temp] "=d" (temp) : [in] "d" ((double)(value)) : ); \
+        __asm__( "fctiw %[temp],%[in]\n\tmfvsrwz %[out],%[temp]\n\t" : [out] "=r" (out), [temp] "=d" (temp) : [in] "d" ((double)(value)) : ); \
         return out;
 
     // FP32 also works with FP64 routine above

